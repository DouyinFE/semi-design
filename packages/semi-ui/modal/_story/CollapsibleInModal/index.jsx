import React, { Component } from 'react';
import { Modal, Collapsible, Button, Table } from '@douyinfe/semi-ui';

class Demo extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen: false,
            visible: false,
            content: null,
            contentIndex: 0
        };
        this.showDialog = this.showDialog.bind(this);
        this.handleOk = this.handleOk.bind(this);
        this.handleCancel = this.handleCancel.bind(this);
        this.toggle = this.toggle.bind(this);
        this.changeContent = this.changeContent.bind(this);
        this.contents = [
            <ul key="ul">
                <li>
                    <p>Semi Design 以内容优先进行设计。</p>
                </li>
                <li>
                    <p>更容易地自定义主题。</p>
                </li>
            </ul>,
            <Table key="table" columns={[
                {
                    title: 'Name',
                    dataIndex: 'name',
                    width: 150,
                    render: (text, record) => text,
                },
                {
                    title: 'Age',
                    dataIndex: 'age',
                    width: 150,
                    render: (text, record) => text,
                },
                {
                    title: 'Address',
                    dataIndex: 'address',
                    render: (text, record) => text,
                },
                {
                    render: (text, record) => (
                        <p>Show Info</p>
                    ),
                    width: 150,
                },
            ]} /> 
        ];
    }
    showDialog() {
        this.setState({
            visible: true,
        });
    }
    handleOk(e) {
        this.setState({
            visible: false,
        });
    }
    handleCancel(e) {
        this.setState({
            visible: false,
        });
    }
    toggle() {
        this.setState({ isOpen: !this.state.isOpen });
    }
    changeContent() {
        this.setState({
            contentIndex: (this.state.contentIndex + 1) % this.contents.length,
        });
    }
    componentDidMount() {
        this.changeContent();
    }

    render() {
        const { isOpen, content, contentIndex } = this.state;
        return (
            <div>
                <Button onClick={this.showDialog}>打开弹窗</Button>
                <Modal
                    title="基本对话框"
                    visible={this.state.visible}
                    onOk={this.handleOk}
                    onCancel={this.handleCancel}
                >
                    <Button onClick={this.toggle}>Toggle</Button>
                    <Button onClick={() => this.changeContent()}>Change Content</Button>
                    <Collapsible isOpen={isOpen}>{this.contents[contentIndex]}</Collapsible>
                </Modal>
            </div>
        );
    }
}
export default () => <Demo />;
