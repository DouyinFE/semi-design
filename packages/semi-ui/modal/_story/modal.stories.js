import React, { useState } from 'react';
import { Select, Modal, Button, Tooltip, Popover } from '../../index';
import CollapsibleInModal from './CollapsibleInModal';
import DynamicContextDemo from './DynamicContext';

export default {
  title: 'Modal',
  parameters: {
    chromatic: { disableSnapshot: true },
  },
}

export {
  CollapsibleInModal,
  DynamicContextDemo
}

const Option = Select.Option;

const DialogComponent = props => {
  const [visible, setVisible] = useState(false);

  const handleOk = e => {
    setVisible(false);
  };

  const handleCancel = e => {
    setVisible(false);
  };

  return (
    <React.Fragment>
      <Button onClick={() => setVisible(true)}>show dialog</Button>
      <Modal
        title="对话框标题"
        visible={visible}
        onOk={handleOk}
        onCancel={handleCancel}
        {...props}
      >
        <input autoFocus />
        <p>basic modal</p>
        <Button onClick={handleCancel}>hide dialog</Button>
      </Modal>
    </React.Fragment>
  );
};

export const Default = () => <DialogComponent />;

Default.story = {
  name: 'default',
};

export const ClickMaskClosableFalse = () => <DialogComponent maskClosable={false} />;

ClickMaskClosableFalse.story = {
  name: 'click mask closable false',
};

function useLoading() {
  const [loading, setLoading] = useState(false);
  function startQuery() {
    setLoading(true);
  }
  return [loading, { startQuery }];
}

function DialogComponentWithLoading() {
  const [loading, actions] = useLoading();
  return <DialogComponent confirmLoading={loading} onOk={actions.startQuery} />;
}

export const WithLoadingTodo = () => <DialogComponentWithLoading />;

WithLoadingTodo.story = {
  name: 'with loading todo',
};

function success() {
  Modal.success('bla bla bla...');
}

function info() {
  Modal.info('info');
}

function error() {
  Modal.error({ title: 'Unfortunately, there is an error', content: 'bla bla bla...' });
}

function warning() {
  Modal.warning({ title: 'Warning: be cautious ahead', content: 'bla bla bla...' });
}

function confirm() {
  Modal.confirm({ title: 'Are you sure ?', content: 'bla bla bla...' });
}

export const ConfirmModal = () => (
  <div>
    <Button onClick={info}>Info</Button>
    <Button onClick={success}>Success</Button>
    <Button onClick={error}>Error</Button>
    <Button onClick={warning}>Warning</Button>
    <Button onClick={confirm}>Confirm</Button>
  </div>
);

ConfirmModal.story = {
  name: 'confirm modal',
};

const Test = () => {
  let modal;

  const showModal = () => {
    modal = Modal.info({
      title: '修改的标题',
      content: (
        <>
          <Button onClick={() => modal && modal.destroy()}>close</Button>
          <Button onClick={() => modal && modal.update({ title: 'lalala updating' })}>
            update
          </Button>
        </>
      ),
    });
  };

  return <Button onClick={() => showModal()}>Info</Button>;
};

export const ModalDestroy = () => <Test />;

ModalDestroy.story = {
  name: 'modal.destroy',
};

const ScrollComponent = props => {
  const [visible, setVisible] = useState(true);

  const handleOk = e => {
    setVisible(false);
  };

  const handleCancel = e => {
    setVisible(false);
  };

  return (
    <div style={{ paddingTop: 900, paddingBotttom: 800 }}>
      <Button onClick={() => setVisible(true)}>show dialog</Button>
      <Modal
        id="modal-test"
        title="对话框标题"
        visible={visible}
        onOk={handleOk}
        onCancel={handleCancel}
        {...props}
      >
        <Select>
          <Option value={1}>opt1</Option>
          <Option value={2}>opt2</Option>
          <Option value={3}>opt3</Option>
          <Option value={4}>opt4</Option>
          <Option value={5}>opt5</Option>
          <Option value={6}>opt6</Option>
        </Select>
        <Tooltip content="fefefefeef">test tooltip in modal</Tooltip>
        <Button onClick={handleCancel}>hide dialog</Button>
        <Popover content={'1,2,3'}>
          <Button>hover</Button>
        </Popover>
      </Modal>
      <Select>
        <Option value={1}>opt1</Option>
        <Option value={2}>opt2</Option>
        <Option value={3}>opt3</Option>
        <Option value={4}>opt4</Option>
        <Option value={5}>opt5</Option>
        <Option value={6}>opt6</Option>
      </Select>
    </div>
  );
};

export const Scroll = () => <ScrollComponent />;

Scroll.story = {
  name: 'scroll',
};

const Popup = () => {
  const [visible, setVisible] = useState(false);
  const getContainer = () => {
    return document.querySelector('.modal-container');
  };
  return (
    <div
      style={{
        height: 320,
        overflow: 'hidden',
        position: 'relative',
        border: '1px solid var(--semi-color-border)',
        borderRadius: 2,
        padding: 24,
        textAlign: 'center',
        background: 'var(--semi-color-fill-0)',
      }}
      className="modal-container"
    >
      <span>Render in this</span>
      <br />
      <br />
      <Button onClick={() => setVisible(true)}>Open Modal</Button>
      <Modal
        title="渲染在指定容器内部"
        visible={visible}
        onCancel={() => setVisible(false)}
        width={200}
        // centered
        // style={{ top: '10px' }}
        getPopupContainer={getContainer}
      >
        <p>This is the content of a basic sidesheet.</p>
        <p>Here is more content...</p>
      </Modal>
    </div>
  );
};

export const RenderInCustomContainer = () => <Popup />;

RenderInCustomContainer.story = {
  name: 'render in custom container',
};

export const KeepDom = () => <DialogComponent keepDOM />;

KeepDom.story = {
  name: 'keepDOM',
};

export const KeepDomNotLazy = () => <DialogComponent keepDOM lazyRender={false} />;

KeepDomNotLazy.story = {
  name: 'keepDOM && not lazy',
};


