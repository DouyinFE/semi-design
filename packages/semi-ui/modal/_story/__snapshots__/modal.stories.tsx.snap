// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`模态框-点击遮罩层不可关闭 1`] = `
<DialogComponent
  maskClosable={false}
>
  <Button
    onClick={[Function]}
  >
    <Button
      block={false}
      disabled={false}
      htmlType="button"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      prefixCls="semi-button"
      size="default"
      theme="light"
      type="primary"
    >
      <button
        aria-disabled={false}
        className="semi-button semi-button-primary semi-button-light"
        disabled={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="button"
      >
        <span
          className="semi-button-content"
          onClick={[Function]}
          x-semi-prop="children"
        >
          show dialog
        </span>
      </button>
    </Button>
  </Button>
  <Button
    onClick={[Function]}
  >
    <Button
      block={false}
      disabled={false}
      htmlType="button"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      prefixCls="semi-button"
      size="default"
      theme="light"
      type="primary"
    >
      <button
        aria-disabled={false}
        className="semi-button semi-button-primary semi-button-light"
        disabled={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="button"
      >
        <span
          className="semi-button-content"
          onClick={[Function]}
          x-semi-prop="children"
        >
          静态调用
        </span>
      </button>
    </Button>
  </Button>
  <Modal
    afterClose={[Function]}
    cancelLoading={false}
    centered={false}
    closable={true}
    closeOnEsc={true}
    confirmLoading={false}
    fullScreen={false}
    hasCancel={true}
    keepDOM={false}
    lazyRender={true}
    mask={true}
    maskClosable={false}
    maskFixed={false}
    motion={true}
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    size="small"
    title="对话框标题"
    visible={false}
    zIndex={1000}
  />
</DialogComponent>
`;

exports[`模态框默认 1`] = `
<DialogComponent>
  <Button
    onClick={[Function]}
  >
    <Button
      block={false}
      disabled={false}
      htmlType="button"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      prefixCls="semi-button"
      size="default"
      theme="light"
      type="primary"
    >
      <button
        aria-disabled={false}
        className="semi-button semi-button-primary semi-button-light"
        disabled={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="button"
      >
        <span
          className="semi-button-content"
          onClick={[Function]}
          x-semi-prop="children"
        >
          show dialog
        </span>
      </button>
    </Button>
  </Button>
  <Button
    onClick={[Function]}
  >
    <Button
      block={false}
      disabled={false}
      htmlType="button"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      prefixCls="semi-button"
      size="default"
      theme="light"
      type="primary"
    >
      <button
        aria-disabled={false}
        className="semi-button semi-button-primary semi-button-light"
        disabled={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="button"
      >
        <span
          className="semi-button-content"
          onClick={[Function]}
          x-semi-prop="children"
        >
          静态调用
        </span>
      </button>
    </Button>
  </Button>
  <Modal
    afterClose={[Function]}
    cancelLoading={false}
    centered={false}
    closable={true}
    closeOnEsc={true}
    confirmLoading={false}
    fullScreen={false}
    hasCancel={true}
    keepDOM={false}
    lazyRender={true}
    mask={true}
    maskClosable={true}
    maskFixed={false}
    motion={true}
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    size="small"
    title="对话框标题"
    visible={false}
    zIndex={1000}
  />
</DialogComponent>
`;
